src/event-handlers/impl/IsaEventHandler.js

```diff
@@ -0,0 +1,107 @@
+import BaseEventHandler from './../base/BaseEventHandler';
+import APP_EVENT from './../../events/base/AppEvent';
+import { IsaService, IsaDtoService } from './../../services';
+import { DEFAULT_STATUS } from './../../constants/';
+import { CONTRACT_AGREEMENT_TYPE } from '@deip/constants';
+
+class IsaEventHandler extends BaseEventHandler {
+  constructor() {
+    super();
+  }
+
+}
+
+const isaEventHandler = new IsaEventHandler();
+
+const isaService = new IsaService();
+const isaDtoService = new IsaDtoService();
+
+isaEventHandler.register(APP_EVENT.CONTRACT_AGREEMENT_PROPOSAL_CREATED, async (event) => {
+
+  const {
+    contractAgreementId,
+    creator,
+    parties,
+    hash,
+    startTime,
+    endTime,
+    type,
+    terms
+  } = event.getEventPayload();
+
+  if (type === CONTRACT_AGREEMENT_TYPE.INCOME_SHARE_AGREEMENT) {
+    await isaService.createIsa({
+      isaId: contractAgreementId,
+      creator,
+      parties,
+      hash,
+      startTime,
+      endTime,
+      type,
+      terms,
+      status: DEFAULT_STATUS.PENDING
+    });
+  }
+
+});
+
+isaEventHandler.register(APP_EVENT.CONTRACT_AGREEMENT_PROPOSAL_DECLINED, async (event) => {
+
+  const { contractAgreementId, type } = event.getEventPayload();
+
+  if (type === CONTRACT_AGREEMENT_TYPE.INCOME_SHARE_AGREEMENT) {
+    await isaService.updateIsa({
+      _id: contractAgreementId,
+      status: DEFAULT_STATUS.REJECTED
+    });
+  }
+});
+
+isaEventHandler.register(APP_EVENT.CONTRACT_AGREEMENT_CREATED, async (event) => {
+
+  const {
+    entityId: contractAgreementId,
+    creator,
+    parties,
+    hash,
+    startTime,
+    endTime,
+    type,
+    terms
+  } = event.getEventPayload();
+
+  if (type === CONTRACT_AGREEMENT_TYPE.INCOME_SHARE_AGREEMENT) {
+    const isa = await isaDtoService.getIsa(contractAgreementId);
+    if (!isa) {
+      await isaService.createIsa({
+        isaId: contractAgreementId,
+        creator,
+        parties,
+        hash,
+        startTime,
+        endTime,
+        type,
+        terms,
+        status: DEFAULT_STATUS.PENDING
+      });
+    }
+  }
+});
+
+isaEventHandler.register(APP_EVENT.CONTRACT_AGREEMENT_ACCEPTED, async (event) => {
+
+  const { entityId: contractAgreementId, party } = event.getEventPayload();
+
+  const isa = await isaDtoService.getIsa(contractAgreementId);
+  if (isa) {
+    const isAllAccepted = isa.chainContract.parties.every((p) => p[1] === 'Accepted')
```

Comment:
Add additional array field to ContractAgreemenSchema called accepteByPartiesand populate this field each time on CONTRACT_AGREEMENT_ACCEPTED event. Define the status by comparing parties and acceptedByParties arrays